src/main/java/mate/academy/internet/shop/dao/jdbc/ProductDaoJdbcImpl.java

```diff
@@ -0,0 +1,104 @@
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Optional;
+import mate.academy.internet.shop.dao.ProductDao;
+import mate.academy.internet.shop.exception.DataProcessingException;
+import mate.academy.internet.shop.lib.Dao;
+import mate.academy.internet.shop.model.Product;
+import mate.academy.internet.shop.util.ConnectionUtil;
+
+@Dao
+public class ProductDaoJdbcImpl implements ProductDao {
+
+    @Override
+    public Product create(Product product) {
+        String query = "INSERT INTO `internet-shop`.`products` "
+                + "(`productname`, `productprice`) VALUES (?, ?)";
+        try (Connection connection = ConnectionUtil.getConnection();) {
+            PreparedStatement preparedStatement = connection.prepareStatement(query);
+            preparedStatement.setString(1, product.getName());
+            preparedStatement.setDouble(2, product.getPrice());
+            preparedStatement.execute();
+        } catch (SQLException e) {
+            throw new DataProcessingException();
+        }
+        return product;
+    }
+
+    @Override
+    public Optional<Product> get(Long id) {
+        String query = "SELECT * FROM products WHERE product_id = ?";
+        try (Connection connection = ConnectionUtil.getConnection()) {
+            PreparedStatement preparedStatement = connection.prepareStatement(query);
+            preparedStatement.setLong(1, id);
+            ResultSet resultSet = preparedStatement.executeQuery();
+            if (resultSet.next()) {
+                long productId = resultSet.getLong("product_id");
+                String productName = resultSet.getString("productname");
+                Double productPrice = resultSet.getDouble("productprice");
+                Product product = new Product(productName, productPrice);
+                product.setProductId(productId);
+                return Optional.of(product);
+            }
+        } catch (SQLException e) {
+            throw new DataProcessingException();
+        }
+        return Optional.empty();
+    }
+
+    @Override
+    public List<Product> getAll() {
+        String query = "SELECT * FROM products";
+        List<Product> productList = new ArrayList<>();
+        try (Connection connection = ConnectionUtil.getConnection()) {
+            PreparedStatement preparedStatement = connection.prepareStatement(query);
+            ResultSet resultSet = preparedStatement.executeQuery();
+            while (resultSet.next()) {
+                long productId = resultSet.getLong("product_id");
+                String productName = resultSet.getString("productname");
+                Double productPrice = resultSet.getDouble("productprice");
+                Product product = new Product(productName, productPrice);
+                product.setProductId(productId);
+                productList.add(product);
+            }
+        } catch (SQLException e) {
+            throw new DataProcessingException();
```

Comment:
```suggestion
            throw new DataProcessingException(e);
```
Do not lose helpful info, re throw the exception

Reply:
done